packages/react/src/KeybindingHint/components/Key.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import VisuallyHidden from '../../_VisuallyHidden'
+import {accessibleKeyName, condensedKeyName, fullKeyName} from '../key-names'
+import type {KeybindingHintFormat} from '../props'
+import {useIsMacOS} from '../../hooks/useIsMacOS'
+
+interface KeyProps {
+  name: string
+  format: KeybindingHintFormat
+}
+
+/** Renders a single key with accessible alternative text. */
+export const Key = ({name, format}: KeyProps) => {
+  const isMacOS = useIsMacOS()
+
+  return (
+    <>
+      <VisuallyHidden>{accessibleKeyName(name, isMacOS)}</VisuallyHidden>
+      <span aria-hidden>{format === 'condensed' ? condensedKeyName(name, isMacOS) : fullKeyName(name, isMacOS)}</span>
```

Comment:
I am curious if this needs to be customisable as well depending on where it is used? 🤔 

For example on IconButton and Tooltip, we would need to append the accessile name to the nae or the description. I wonder how this visually hidden element will play out here? 👀

Reply:
I have verified that if we are generating the accessible name/description from DOM, this results in the correct output. For example, if you render this inside of a `Button` element, the `VisuallyHidden` part is what becomes part of the accessible name while the `aria-hidden` part is ignored. This is what we want since the `VisuallyHidden` part contains the readable string version of the key icons. 

For when we are appending the shortcut to a plain string used as `aria-label` or `aria-description`, we expose a `getAccessibleKeybindingHintString` that allows us to skip the whole DOM part and just get a usable string, so that case is covered as well.

Reply:
Wonderful! I have played with it as well and it works great ✨ 

I have one comment though regarding using this in Tooltip:

I quickly implemented it to see how that would work and looks in Tooltip and I think we need to work through the colours if this is how we are going to use this component in Tooltip

<img width="248" alt="Tooltip is open on a button and it has keyboard shortcuts on the tooltip. The text colour of the keybinding component doesn't create enough contrast on the tooltip background " src="https://github.com/user-attachments/assets/6b5e7c39-85af-4d4e-ba8d-18ea7cb20820">

tagging @mperrotti here for design 👀

Reply:
Regarding using `getAccessibleKeybindingHintString` ,  is there a way to handle determining `isMacOS` under the hood so that we don't worry about it every time when we use this function? 

Let me know if there is anything I am missing here

Reply:
This is actually one of the reasons the `onEmphasis` variant exists - we'd definitely want to use that here

Reply:
>I quickly implemented it to see how that would work and looks in Tooltip and I think we need to work through the colours if this is how we are going to use this component in Tooltip

I think this should be resolvable with just `variant="onEmphasis"`, which should flip the text color to work here. 



> Regarding using `getAccessibleKeybindingHintString` , is there a way to handle determining `isMacOS` under the hood so that we don't worry about it every time when we use this function?

Not really, since the macOS logic is now inside a hook - we'd need to make this a hook as well. There are rare cases where we call this outside the react app, so we'd rather not have this as a hook. 

What we could do is introduce a new hook wrapper around this like `useAccessibleKeybindingHintString`

Reply:
No worries, let's keep it as is 👍🏻